### ISSUE TITLE
[a11y]: Site switcher status (opened/closed or expanded/collapsed) is not available to JAWS and other assisteive technology

### ISSUE THREAD

**Issue by philljenkins** [carbon-design-system/carbon#14136]
### Package

@carbon/react

### Browser

Chrome

### Operating System

Windows

### Package version

v11

### Automated testing tool and ruleset

IBM Equal Access Checker 3.1.51

### Assistive technology

JAWS 2023

### Description

When encountering the "Switch sites" icon in many IBM pages and many apps, the status of whether the area is open/expanded or not is not announced by the screen reader.  The JAWS user gets no feedback as to what happened when the button was pressed. 

The button is well-labeled, and the region is well-labeled, but the change is not labeled or announced.

### WCAG 2.1 Violation

[4.1.2 Name, Role, Value](https://www.ibm.com/able/requirements/requirements/#4_1_2)

### Reproduction/example

Run JAWS on any pages on http://www.ibm.com/able

### Steps to reproduce

1. Load JAWS
2. Open ibm.com/able
3. Tab to the "Switch Site" button in the header
4. Click the button, no announcement is made

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by philljenkins** [comment]
Original accessibility issue was opened against the IBM Accessibility toolkit https://github.ibm.com/IBMa/E2E/issues/4711

**Comment by tw15egan** [comment]
Would adding `aria-expanded` based on the state fix this issue? I believe @guidari just added this as part of #13894 ([line](https://github.com/carbon-design-system/carbon/pull/13894/files#diff-bbc089d28b960d88ad2b20deedd14ddee053ca23314a5e6df8562691d8ee0095R614))

**Comment by guidari** [comment]
Yeah! The `aria-expanded` should work. When it clicked the button here in [this story ](http://localhost:3000/?path=/story/components-ui-shell-header--header-w-actions-and-switcher) it says, "Close switcher" or "Open switcher" depending on the state. I'm not sure if that is the issue you are facing.

**Comment by tay1orjones** [comment]
Need to confirm this reads correctly in JAWS, if so we'll close this out

**Comment by philljenkins** [comment]
Is there a way (e.g. sandbox, etc.) for me to test JAWS with the proposed fix?

**Comment by tw15egan** [comment]
@philljenkins It will eventually, but its dependent on the `gatsby-theme-carbon` being updated to the latest version of Carbon. You can verify the changes here: https://deploy-preview-14196--carbon-components-react.netlify.app/?path=/story/components-ui-shell-header--header-w-actions-and-switcher

**Comment by guidari** [comment]
I just tested on Jaws and it is working as expected.
